src/snowflake/cli/plugins/object/common.py

```diff
@@ -0,0 +1,81 @@
+from dataclasses import dataclass
+from typing import Optional
+
+import typer
+from click import ClickException
+from snowflake.cli.api.project.util import (
+    QUOTED_IDENTIFIER_REGEX,
+    UNQUOTED_IDENTIFIER_REGEX,
+    is_valid_identifier,
+    to_string_literal,
+)
+
+
+@dataclass
+class Tag:
+    name: str
+    value: str
+
+    def __post_init__(self):
+        if not is_valid_identifier(self.name):
+            raise ValueError("name of a tag must be a valid snowflake identifier")
+
+    def value_string_literal(self):
+        return to_string_literal(self.value)
+
+
+class TagError(ClickException):
+    def __init__(self):
+        super().__init__(
+            "tag must be in the format <name>=<value> where 'name' is a valid identifier and value is a string"
+        )
+
+
+def _parse_tag(tag: str) -> Tag:
```

Comment:
For the tag issue https://docs.google.com/document/d/1qsf2igvHGxVFWvN1f52ryrC-6cFrfTSaa84sIXR8nQ0/edit 
Though you see some warnings, the service creation succeeds. Could you please check service existence and use `select system$get_tag('<tag_key>', '<service_name>', 'service');` to verify the tag is set successfully?

Reply:
I have confirmed that the tag is set correctly. I was just thinking that it's not ideal to have unexplained warnings displayed to the user, but the feature is functional.